File: src/main/java/me/mintytc/azapi/api/interfaces/Mergeable.java
package me.mintytc.azapi.api.interfaces;

/**
 * @since 1.0.0-R0.1-BASE
 *
 */
public interface Mergeable<T> {
	/**
	 * Merge this object with another of the same type.
	 *
	 * @param other the other object
	 * @return a new merged instance
	 */
	T merge(T other);
}

File: src/main/java/me/mintytc/azapi/interfaces/Mergeable.kt
package me.mintytc.azapi.interfaces

/**
 * @since 1.0.0-R0.1
 */
interface Mergeable<T> {
	/**
	 * Merge this object with another of the same type.
	 *
	 * @param other the other object
	 *
	 * @return a new merged instance
	 */
	fun merge(other: T?): T?
}